libs/wingc/src/ast.rs

```diff
@@ -117,32 +117,36 @@ impl Display for Type {
 			Type::MutMap(t) => write!(f, "MutMap<{}>", t),
 			Type::Set(t) => write!(f, "Set<{}>", t),
 			Type::MutSet(t) => write!(f, "MutSet<{}>", t),
-			Type::FunctionSignature(sig) => {
-				let phase_str = match sig.flight {
-					Phase::Inflight => "inflight ",
-					Phase::Preflight => "preflight ",
-					Phase::Independent => "",
-				};
-				let params_str = sig
-					.parameters
-					.iter()
-					.map(|a| format!("{}", a))
-					.collect::<Vec<String>>()
-					.join(", ");
-				let ret_type_str = if let Some(ret_val) = &sig.return_type {
-					format!("{}", ret_val)
-				} else {
-					"void".to_string()
-				};
-				write!(f, "{phase_str}({params_str}): {ret_type_str}",)
-			}
+			Type::FunctionSignature(sig) => write!(f, "{}", sig),
```

Comment:
I don't feel like this is duplication, our naming just makes it seem like it.
One of these corresponds to the AST node for a function type annotation, the other is our representation of a function type in our type system. 

The `Type` used in this file should probably be renamed to `TypeAnnotation` or something to help with that.

Reply:
Ah great idea, I like `TypeAnnotation` 👍